index_batcher.go

```diff
@@ -0,0 +1,164 @@
+package bleve
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NOTE XXX As this approach uses carefully orchestrated interactions
```

Comment:
I think we should rewrite this comment with some concrete explanations.  Just saying 'be careful' isn't helpful if users aren't sure what the issues are.